pkg/controller/hawtio/certificate.go

```diff
@@ -83,3 +86,64 @@ func generateCertificateSecret(name string, namespace string, caSecret *corev1.S
 		}, Type: corev1.SecretTypeTLS,
 	}, nil
 }
+
+func ValidateCertificate(caSecret corev1.Secret, validAtLeastForHours float64) bool {
+	block, _ := pem.Decode(caSecret.Data[corev1.TLSCertKey])
+	cert, err := x509.ParseCertificate(block.Bytes)
+
+	if err != nil {
+		log.Error(err, "certificate reading error")
+	}
+
+	diff := cert.NotAfter.Sub(time.Now()).Hours()
+	// if cert is valid longer than certain amount of hours
+	if diff > validAtLeastForHours {
+		log.Info(fmt.Sprintf("Certificate is valid for %.0f days", diff/24))
+		return true
+	}
+	//if is valid
+	return false
+}
+
+func createCertValidationCronJob(name, namespace, hawtioName, image string) *v1beta1.CronJob {
+	cronjob := &v1beta1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1beta1.CronJobSpec{
+			Schedule:          "* */12 * * *",
+			ConcurrencyPolicy: v1beta1.ForbidConcurrent,
+			JobTemplate: v1beta1.JobTemplateSpec{
+				Spec: v1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							ServiceAccountName: "hawtio-operator",
+							RestartPolicy:      "Never",
+							Containers: []corev1.Container{
+								{
+									Name:  "hawtio-operator",
+									Image: image,
+									Command: []string{
+										"hawtio-operator",
+									},
+									Args: []string{
+										"cert-expiry-check",
+									},
+									Env: []corev1.EnvVar{
+										{
+											Name:  "WATCH_NAMESPACE",
```

Comment:
I'd rather turn this into an option of the `cert-expiry-check` command, rather than an environment variable.